internal/engine/module_builder.go

```diff
@@ -315,52 +315,91 @@ func (b *ModuleBuilder) GetConfigDoc(value cue.Value) ([][]string, error) {
 		return nil, fmt.Errorf("lookup %s failed: %w", apiv1.ConfigValuesSelector, cfgValues.Err())
 	}
 
-	labelDomain := regexp.MustCompile(`^([a-zA-Z0-9-_.]+)?(".+")?$`)
+	rows, err := iterateFields(cfgValues)
+	if err != nil {
+		return nil, err
+	}
+
+	return rows, nil
+}
 
+func iterateFields(v cue.Value) ([][]string, error) {
 	var rows [][]string
-	configDataInfo := func(v cue.Value) bool {
-		var row []string
-		var noDoc bool
-		var doc string
-
-		for _, d := range v.Doc() {
-			if line := len(d.List) - 1; line >= 0 {
-				switch d.List[line].Text {
-				case "// +nodoc":
-					noDoc = true
-					break
-				}
+
+	fields, err := v.Fields(
+		cue.Optional(true),
+		cue.Concrete(true),
+		cue.Docs(true),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("Cue Fields Error: %w", err)
+	}
+
+	for fields.Next() {
+		v := fields.Value()
+		fmt.Println(fields.Label(), v.IncompleteKind(), v.IsConcrete())
+
+		// We are chekcing if the field is a struct and not optional and is concrete before we iterate through it
+		// this allows for definition of default values as full structs without generating output for each
+		// field in the struct where it doesn't make sense e.g.
+		//
+		// - annotations?: {[string]: string}
+		// - affinity: corev1.Affinity | *{nodeAffinity: requiredDuringSchedulingIgnoredDuringExecution: nodeSelectorTerms: [...]}
+		if v.IncompleteKind() == cue.StructKind && !fields.IsOptional() && v.IsConcrete() {
```

Comment:
Do we add another convention, like `+nodoc`, that would force rendering of `metadata: labels:`?

With this check for struct and optional we loose `metadata: labels:` from the output.

Reply:
Having `+nodoc` sounds great!

Reply:
Sorry, I'll expand on this.

So this already supports `+nodoc`, the issue I've hit which brought this up is that `metadata: labels:` has data in it and so the generated output doesn't contain `metadata: labels:` if the data in it has `+nodoc` set.

What I'm thinking to do is add another `+` label that would force adding a field to the output instead of diving into it if it's an object. Something like `+value`, not sure of anything already existing that would make sense.

Not at my machine at the moment but will put together an example to show you when I can.

Reply:
Ah I get now what you are saying, assuming the parent struct has `+nodoc` there are cases when we want to include some of the struct fields. In this case we could use `+doc` that would take precedence over the parent exclusion.

Reply:
Actually the other way around, we want to have the labels field documented but not the labels in it. By having those labels marked `+nodoc` the label field is not documented.

Reply:
So for example we have the following in the timoni schema:

```cue
// Metadata defines the schema for Kubernetes object metadata.
#Metadata: {
        ...
        // Map of string keys and values that can be used to organize and categorize (scope and select) objects.
        // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
        labels: #Labels

        // Standard Kubernetes labels: app name, version and managed-by.
        labels: {
                "\(#StdLabelName)":      name
                "\(#StdLabelVersion)":   #Version
                "\(#StdLabelManagedBy)": "timoni"
        }

        // LabelSelector selects Pods based on the app.kubernetes.io/name label.
        #LabelSelector: #Labels & {
                "\(#StdLabelName)": name
        }
}
```

```markdown
| `metadata: labels: app.kubernetes.io/name` | `testing` |
| `metadata: labels: app.kubernetes.io/version` | `vX.Y.Z` |
| `metadata: labels: app.kubernetes.io/managed-by` | `timoni` |
```

We then update this with `+nodoc` and as follows:

```cue
// Metadata defines the schema for Kubernetes object metadata.
#Metadata: {
        ...
        // Map of string keys and values that can be used to organize and categorize (scope and select) objects.
        // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
        labels: #Labels

        // Standard Kubernetes labels: app name, version and managed-by.
        labels: {
                // +nodoc
                "\(#StdLabelName)":      name
                // +nodoc
                "\(#StdLabelVersion)":   #Version
                // +nodoc
                "\(#StdLabelManagedBy)": "timoni"
        }

        // LabelSelector selects Pods based on the app.kubernetes.io/name label.
        #LabelSelector: #Labels & {
                "\(#StdLabelName)": name
        }
}
```

The labels will not be part of the output now but neither will `metadata: labels:` because it was entered into as a structure field. I'll take another look at this and see if I can sort out this issue without needing another keyword. But if not another keyword would be the approach I'm thinking of at the moment. Which would remove the need to have `+nodoc` on each label as well e.g.

```cue
// Metadata defines the schema for Kubernetes object metadata.
#Metadata: {
        ...
        // Map of string keys and values that can be used to organize and categorize (scope and select) objects.
        // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
        // +doc
        labels: #Labels

        // Standard Kubernetes labels: app name, version and managed-by.
        labels: {
                "\(#StdLabelName)":      name
                "\(#StdLabelVersion)":   #Version
                "\(#StdLabelManagedBy)": "timoni"
        }

        // LabelSelector selects Pods based on the app.kubernetes.io/name label.
        #LabelSelector: #Labels & {
                "\(#StdLabelName)": name
        }
}
```

```markdown
| `metadata: labels: ` | `{"app.kubernetes.io/name": "testing",...}` |
```

Reply:
I guess I could check the output of the function to see if there are any rows returned and if not then put that field in place, could avoid needing a new `+` tag though would need `+nodoc` on all fields in a structure. Will test this tomorrow.

Reply:
@stefanprodan I've updated this so that it will now add a row for any struct that has `+nodoc` on all child fields. Not sure if this is the best approach as it means it needs to be structured in a certain way e.g.

```cue
// Metadata defines the schema for Kubernetes object metadata.
#Metadata: {
        ...
        // Map of string keys and values that can be used to organize and categorize (scope and select) objects.
        // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
        labels: #Labels

        // Standard Kubernetes labels: app name, version and managed-by.
        // +nodoc
        labels: "\(#StdLabelName)": name
}
```

This will not do what we want as `+nodoc` is being applied to `labels` (I believe), it would need to be structured like this:

```cue
// Metadata defines the schema for Kubernetes object metadata.
#Metadata: {
        ...
        // Map of string keys and values that can be used to organize and categorize (scope and select) objects.
        // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
        labels: #Labels

        // Standard Kubernetes labels: app name, version and managed-by.
        labels: {
                // +nodoc
                "\(#StdLabelName)":      name
        }
}
```